Publish minutes of 2022-04-14 meeting (#202)
hober/web-extensions: _minutes/2022-04-14-wecg.md

 * [dave] Prerendered has not shipped yet, we do not know yet how many extensions actually break.
 * [timothy] I was of the impression that Chrome already had some prerendering support.
 * [dave] Prerendering was shipped and unshipped on desktop. Our goal is that extensions will execute in prerender.
 * [rob] Let's continue the rest of the discussion in the Github issue or in the Matrix chat since we've already spent half of the meeting on this.

[Issue 191](https://github.com/w3c/webextensions/issues/191): Should background pages be allowed to navigate?

 * [timothy] I filed this. There's some inconsistency across browsers. Safari allows this, Chrome blocks it, Firefox seems to allow it.
 * [tomislav] Officially not by design in Firefox, we just didn't prevent it. Safe to say it's not something we intend to support.
 * [rob] What about same origin navigation?
 * [timothy] That's the only thing we allow in Safari.
 * [rob] In the issue you mentioned “Safari 14+ allows navigations” - does that mean same-origin navigation only and not remote URLs?
 * [timothy] Yes.

>  * [simeon] Already covered what Chrome does here, but

Reviewer: @dotproto Did you have more thoughts that you said for this line?

Reviewer 2: Thanks for tagging this, I'm updating this comment as part of my review.

 * [rob] For Firefox we would limit this to MV3, maybe also MV2 if it makes sense later. Chrome does not support background pages/event pages in MV3. How about Safari?
 * [timothy] In Safari we do support this currently, but don't anticipate much breakage.
 * **Resolution**: No, background pages should not be allowed to navigate to remote URLs. Same-origin navigation are permitted.

[Issue 196](https://github.com/w3c/webextensions/issues/196): Proposal: onDoubleClicked for browserAction

 * [timothy] Short recap: some extension developers implement double-click behavior to change what the extension does based on whether the user single or double clicks the action.
 * [tomislav] Not in favor, because there are no other buttons in the browser UI with similar behavior. There is no affordance for it. And it is not available on mobile either. Probably not going to implement that in Firefox.
 * [simeon] Hesitant about the feature too, but understands the developer's concern. This is already possible in user land (with the existing APIs), so there is probably no need for a new platform API for this.

>  * [rob] Chrome's documentation emphasizes that setTimeout/setInterval are not reliable, but they are actually for this specific use case.

Reviewer: Resolution was to close this and not add a double click event.

 12. Alexei (Privacy Badger)
 13. Zane Bond (Keeper Security)
 14. Krzysztof Modras (Ghostery)
 15. Mukul Purohit (Microsoft)
 16. Tyler Carson (Keeper)
 17. Richard Worth (Capital One)
 18. James Hycner (Keeper)

## Meeting notes

[Issue 8](https://github.com/w3c/webextensions/issues/8#issue-917621223): executeScript API may inject scripts in an unexpected target at runtime (documentID proposal)

 * [timothy] This is one of the first issues that mentioned the concept of documentID. Dave from the Chrome team is here to discuss.
 * [dave] Chrome is currently working on pre-rendering. Initial motivation for pursuing documentId is that frameID is a bit difficult to work with in this context.
 * [rob] This is the “Exposing Document Info to Extensions” proposal that is being discussed: https://docs.google.com/document/d/1xOe5e8CDU9HM6sazGg6aqvK0iPnptilmwBS9qiZSfkY/edit
 * [dave] Proposing documentId, parentDocumentId, documentLifeCycle (prerendering, cached, etc.) and frameType properties. frameType includes fenced_frame (that we're working on as part of Privacy Sandbox).
 * [dave] Properties on most events, but only documentId in the runtime.onMessage events for performance reasons.

>  * [rob] runtime.onMessage is less frequent than webRequest/webNavigation, but there are more fields on webRequest/webNavigation events than messaging.

Reviewer 2: ```suggestion
 * [rob] runtime.onMessage is less frequent than webRequest/webNavigation, but there are more fields on webRequest/webNavigation events than messaging.
 * [dave] Intent was to limit the amount of data exposed in critical paths, will defer to defer to area experts on what should be exposed where.
```

I think this captures the spirit of Dave's response.

 * [dave] Still WIP, but content scripts would execute in prerendered pages. executeScript+allFrames currently executes in the active page, but should also run in prerendered pages.
 * [tomislav] There will be situations where there are two documents in the top-level frame.
 * [tomislav] For runtime.onMessage/onConnect it could be useful for extensions to know whether the caller is in a prerender or not - suggesting to include more properties besides documentId (e.g. documentLifeCycle).
 * [dave] That comes down to the feedback of whether we expose all the properties on the various events.
 * [rob] documentId is currently documented to be some opaque string. What about non-document contexts such as web workers? They can initiate requests
 * [dave] it's an optional flag. documentId is only exposed for things backed by frames. I'll need to think about that as well.
 * [alexei] This proposal may help with answering questions like what is the top-level document URL that a request belongs to, so that's exciting. This however does not address the cases where a Service worker fetches on behalf of a web page.

>  * [dave] Correct, subsequent request not executed in the context of the service worker

Reviewer 2: ```suggestion
 * [dave] Correct, subsequent requests are not executed in the context of the service worker
```

 * [dave] Prerendered has not shipped yet, we do not know yet how many extensions actually break.
 * [timothy] I was of the impression that Chrome already had some prerendering support.
 * [dave] Prerendering was shipped and unshipped on desktop. Our goal is that extensions will execute in prerender.
 * [rob] Let's continue the rest of the discussion in the Github issue or in the Matrix chat since we've already spent half of the meeting on this.

[Issue 191](https://github.com/w3c/webextensions/issues/191): Should background pages be allowed to navigate?

 * [timothy] I filed this. There's some inconsistency across browsers. Safari allows this, Chrome blocks it, Firefox seems to allow it.
 * [tomislav] Officially not by design in Firefox, we just didn't prevent it. Safe to say it's not something we intend to support.
 * [rob] What about same origin navigation?
 * [timothy] That's the only thing we allow in Safari.
 * [rob] In the issue you mentioned “Safari 14+ allows navigations” - does that mean same-origin navigation only and not remote URLs?
 * [timothy] Yes.

>  * [simeon] Already covered what Chrome does here, but

Reviewer 2: ```suggestion
 * [simeon] We have already covered what Chrome does here (not allowed), but I'm concerned about the security impact of allowing background page navigation. Seems like this might introduce unnecessary risk.
```

 * [rob] Related, what about extension popups? They can navigate elsewhere to a remote URL, is this support deliberate?
 * [timothy] Pretty sure that Chrome supports this too, we checked and implemented it too.

>  * [simeon] Not sure if it is deliberate. I can see the use case for same origin navigation, but I'm hesitant about potential security issues and encouraging (or at least implicitly endorsing patterns we don't want to encourage.

Reviewer 2: ```suggestion
 * [simeon] Not sure if it is deliberate. I can see the use case for same origin navigation, but I'm hesitant about potential security issues and encouraging (or at least implicitly endorsing) patterns we don't want to perpetuate.
```

>  * [carlos] If we are resolving to change behavior, suggest limiting this to a new manifest version bump or at least limiting it to manifest v3.

Reviewer 2: ```suggestion
 * [carlos] If we are resolving to change behavior, suggest limiting this to a new manifest version bump or at least limiting it to manifest v3.
 * [simeon] How we approach breaking changes has come up in a few contexts. I will create a placeholder issue to discuss this in more detail. (post-meeting follow up: #200)
```

 * [timothy] Open to this on the Safari side
 * [rob] Are you asking about triggering CSP violation events (report-only) or reporting them to a remote server? I have concerns about the latter.
 * [carlos] How do the browsers feel about this?
 * [timothy] Initially supportive. (Safari)
 * [simeon] I think that this worth further discussion. (Chrome)
 * [rob] I'm not convinced that we need this capability, but interested in following along with the discussion. (Firefox)

[Issue 198](https://github.com/w3c/webextensions/issues/198): Proposal: action.default_area=navbar and action.default_area=hidden

 * [carlos] Given the time, best to move on to 199. This is a sub-issue of that one.

[Issue 199](https://github.com/w3c/webextensions/issues/199): Streamline and optimise after-install dialog action buttons

 * [carlos] Inconsistent behavior across browsers in the install flow. Would be nice to offer controls to allow the user to e.g. pin the extension action button.
 * [rob] We can talk about it, but this isn't something we'd standardize because this is browser-specific UI and UX.

>  * [simeon] Worth mentioning that Chrome is working on updating the (extension) browser UI and toolbar. We intentionally passed on the ability to pin the button in the install flow. I will follow up to that in the issue.

Reviewer 2: ```suggestion
 * [simeon] Worth mentioning that Chrome is working on updating the (extension) browser UI and toolbar. We intentionally passed on the ability to pin the button in the install flow. I will follow up on that in the issue.
```
